server/api/org/update/controller.js

```diff
@@ -0,0 +1,98 @@
+import Joi from 'joi';
+import Boom from 'boom';
+import compose from 'koa-compose';
+import packJSONRPC from '../../../middleware/packJSONRPC';
+import { validateRequest } from '../../../middleware/validation';
+import {
+  decorateSession,
+  isUserAuthenticated,
+  decorateUserPermissions,
+  findUserPermissionIndex,
+} from '../../../middleware/auth';
+import updateOrg, { checkOrgExists } from './service';
+import { AuthorizationError, OrgNotFoundError } from '../../../constants/errors';
+
+export const validationSchema = {
+  body: {
+    id: Joi.string()
+      .length(24)
+      .hex()
+      .required(),
+    name: Joi.string()
+      .trim()
+      .min(3)
+      .max(100)
+      .optional(),
+    slug: Joi.string()
+      .lowercase()
+      .trim()
+      .min(3)
+      .max(30)
+      .optional()
+      .regex(/^[A-Za-z0-9\-_]*$/, { name: 'slug' }),
+  },
+};
+
+const doesRequestedOrgExist = async (ctx, next) => {
+  const { request } = ctx;
+  const org = await checkOrgExists(ctx, request.body.id);
+
+  if (!org) {
+    throw new OrgNotFoundError(`Org ${request.body.id} cannot be found`);
+  }
+
+  await next();
+};
+
+const isUserAuthorized = async ({ request }, next) => {
+  const hasPermission = findUserPermissionIndex(request.session.user.permissions, {
+    name: 'yeep.org.write',
+    orgId: request.body.id,
```

Comment:
You should also check for `null` orgId, a.k.a. _global_ scope. Please refer to some other controller to see how we deduplicate orgs using `Set`.